sdk/storage/Azure.Storage.Common/src/Shared/PartitionedUploader.cs

```diff
@@ -376,5 +440,169 @@ public struct Behaviors
                 partition.Dispose();
             }
         }
+
+        /// <summary>
+        /// Some streams will throw if you try to access their length so we wrap
+        /// the check in a TryGet helper.
+        /// </summary>
+        private static long? GetLengthOrDefault(Stream content)
+        {
+            if (content == null)
+            {
+                /* Returning 0 instead of default puts us on the quick and clean one-shot upload,
+                 * which produces more consistent fail state with how a 1-1 method on the convenience
+                 * layer would fail.
+                 */
+                return 0;
+            }
+            try
+            {
+                if (content.CanSeek)
+                {
+                    return content.Length;
```

Comment:
I mean, someone can write code that can do that. `Stream` documentation is very explicit that length, position, and seekability are an all-or-nothing package, and so `CanSeek` is the indicator. Anyone who gives us a stream that behaves otherwise is breaking the .NET API contract.

Reply:
Yeah. I'm assuming Stream implemented according to .NET guidance. Thank you!